lib/zlib.js

```diff
@@ -0,0 +1,9 @@
+'use strict'
+const semver = require('semver')
+
+module.exports = select(process.version)
+module.exports._SELECT_ZLIB = select
+
+function select (version) {
+  return semver.gt(version, 'v9.0.0-0') ? require('zlib') : require('minizlib')
```

Comment:
Probably could have also done something like `parseInt(version.slice(1), 10) >= 9` to avoid having to pull in `semver`?

Reply:
Considering how widely-used `semver` is, I fully expect this to just be a deduped dep, and I'd much rather have semver-awareness here. I don't think this will actually have much of an effect on the dependency tree, in the end.